src/uct/sm/scopy/base/scopy_ep.c

```diff
@@ -172,17 +137,28 @@ ucs_arbiter_cb_result_t uct_scopy_ep_progress_tx(ucs_arbiter_t *arbiter,
         status   = iface->tx(&ep->super.super, tx->iov, tx->iov_cnt,
                              &tx->iov_iter, &seg_size, tx->remote_addr,
                              tx->rkey, tx->op);
-        status = uct_scopy_ep_tx_comp(ep, tx, status, seg_size);
-        if (ucs_likely(!UCS_STATUS_IS_ERR(status))) {
-            count++;
-            if (status == UCS_INPROGRESS) {
+        if (!UCS_STATUS_IS_ERR(status)) {
+            (*count)++;
+            tx->remote_addr += seg_size;
+            uct_scopy_trace_data(tx);
+
+            if (tx->iov_iter.iov_index <= tx->iov_cnt) {
```

Comment:
should it be < ?
seems we better remove the operation and decrement outstanding, call completion etc. in case of error

Reply:
> should it be < ?

yes, it should. good catch.

> seems we better remove the operation and decrement outstanding, call completion etc. in case of error

not sure that I got this comment

Reply:
we decrement outstanding, call completion etc. in case of error or when all data writen/read.

Reply:
error handling seems wrong - we don't update outstanding counter, flush may hang?

Reply:
if an error occurred, it will be done below (150-151 lines)